src/vcpkg/build.cpp

```diff
@@ -396,6 +397,28 @@ namespace vcpkg::Build
                 if (auto p_val = val.get()) env.emplace(var, *p_val);
             }
 
+            /*
+             * On Windows 10 (>= 8.1) it is a user-friendly way to automatically set HTTP_PROXY and HTTPS_PROXY
+             * environment variables by reading proxy settings via WinHttpGetIEProxyConfigForCurrentUser, preventing users
+             * set and unset these variables manually (which is not a decent way).
+             * It is common in China or any other regions that needs an proxy software (v2ray, shadowsocks, etc.), which
+             * sets the IE Proxy Settings, but not setting environment variables. This will make vcpkg easier to use,
+             * specially when use vcpkg in Visual Studio, we even cannot set HTTP(S)_PROXY in CLI, if we want to open or
+             * close Proxy we need to restart VS.
+             */
+            if (System::get_windows_ie_proxy_enabled())
+            {
+                auto proxy = System::get_windows_ie_proxy_server();
+                if (proxy.has_value())
+                {
+                    // Most HTTP proxy DOES proxy HTTPS requests.
+                    env.emplace("HTTP_PROXY", proxy.value().c_str());
+                    env.emplace("HTTPS_PROXY", proxy.value().c_str());
```

Comment:
I'm still a bit concerned about setting the S one of these given that I observed failures to connect in testing. Can you point to a means of testing this?

Reply:
Sorry, it seems that I got confused about HTTP and HTTPS proxy. HTTP proxies can proxy https:// requests and a HTTPS proxy is dedicated to provide end to end security while HTTP proxy is not.